.github/actions/setup-binary/action.yml

```diff
@@ -0,0 +1,135 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'Setup Binary'
+
+description: |-
+  Use this action to install and cache a binary for use within a GitHub workflow. This action wraps the functionality of go-getter to provide advanced donwload and checksum capabilities.
+
+inputs:
+  download_url:
+    description: 'The download URL for the binary. This must a valid go-getter URL. See https://github.com/hashicorp/go-getter#url-format'
+    required: true
+  checksum:
+    description: 'The checksum for the downloaded artifact. See https://github.com/hashicorp/go-getter#checksumming'
+    required: false
+  install_path:
+    description: 'The path on disk in which to install the binary (defaults to the current directory). This includes the folder and binary name (e.g. /path/to/my/binary). This action does not modify $PATH; if you want the binary to be accessible, make sure the parent folder is in $PATH.'
```

Comment:
I think you need to say "this must be an absolute path" since go-getter requires that.

Reply:
So I tried with relative paths e.g. ./test and it creates the dir and works. I think the install dir cant/shouldnt be the CWD.

Reply:
Ah I see. That's interesting.